src/MessageFormatterInterface.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace GuzzleHttp;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+interface MessageFormatterInterface
+{
+    /**
+     * Apache Common Log Format.
+     *
+     * @link https://httpd.apache.org/docs/2.4/logs.html#common
+     *
+     * @var string
+     */
+    public const CLF = "{hostname} {req_header_User-Agent} - [{date_common_log}] \"{method} {target} HTTP/{version}\" {code} {res_header_Content-Length}";
+    public const DEBUG = ">>>>>>>>\n{request}\n<<<<<<<<\n{response}\n--------\n{error}";
+    public const SHORT = '[{ts}] "{method} {target} HTTP/{version}" {code}';
```

Comment:
Although I am 👍 for this.. it might be a BC for some ppl that might have been extending the `MessageFormatter` class implementation and overriding these constants.. now they wont be able to.

Reply:
I supposed that `MessageFormatter` will be marked as final and thus extending will be prohibited.

Reply:
Yes.. but we add the final annotation and not the final php keyword in order not to hard enforce this rule on 7. So developers can still extend this class. We will enforce the final on v8.. on v8 we could move these to the interface as you did..

Reply:
Ok. For v7 I move consts back to `MessageFormatter`, ok?

Reply:
I'd say put them in both, but annotate the ones in the class as deprecated.

Reply:
https://3v4l.org/hbjIH

Reply:
Oh, if you put them only in the interface, are they still accessible as if they were on the class, then, so not a BC break?

Reply:
It is a breaking change for ppl that use to extend the MessageFormatter class.

Reply:
I moved format consts back into `MessageFormatter` and annotated they will be moved in v8.

Reply:
Is it? https://3v4l.org/acm6d

Reply:
https://3v4l.org/nipM9

Reply:
after the move to the interface ppl will have problem

Reply:
W8 W8.. my previous code is wrong...

Out of PHP documentation:

> It's possible for interfaces to have constants. Interface constants work exactly like class constants except they cannot be overridden by a class/interface that inherits them. 

but check this:

https://3v4l.org/qrZmn

why is this working 🤔

Reply:
:trollface: D'no

Reply:
So if it's a first level child it works as expected https://3v4l.org/ctK02

Reply:
Concluding this is a PHP bug https://bugs.php.net/bug.php?id=73348